.github/workflows/build_matrix.yml

```diff
@@ -38,6 +38,6 @@ jobs:
       config: ${{ matrix.config }}
       os: ${{ matrix.os }}
       platform: ${{ matrix.platform }}
-      ref: ${{ matrix.ref }}
+      ref: ${{ inputs.ref }}
```

Comment:
this was the typo that made the whole thing silently build the current commit, rather than the target ref.